src/components/TextInput/TextInput.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import ShowError from "../ShowError/ShowError";
+import { getTasks, confirmTaskEditing } from "../../service/Requests";
+
+import done from '../../img/done.svg';
+import close from '../../img/close.svg';
+import './style.scss';
+
+const TaskEditing = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [tasks, setTasks] = useState([]);
+  const [error, setError] = useState('');
+  const [text, setText] = useState('');
+
+  useEffect(() => {
+    const getAllTasks = async () => {
+      const resp = await getTasks();
+      if (resp.statusText === 'OK') {
+        setTasks(resp.data);
+      }
+    };
+    getAllTasks();
+  }, []);
+
+  useEffect(() => {
+    for (let i = 0; i < tasks.length; i++) {
+      if (tasks[i]._id === id) {
+        setText(tasks[i].text);
+      }
+    }
+  }, [tasks])
+
```

Comment:
First, useEffect does not write async/await.
Secondly, 2 useEffect is also not written.
Thirdly, why do you get all the tasks when you have to work with only one?